src/test/java/net/helpscout/api/WebhookApiTest.java

```diff
@@ -0,0 +1,109 @@
+     * x-forwarded-server: youcallback.domain.com
+     * connection: Keep-Alive
+     *
+     * Pay attention to case of header names - according to specification
+     * header names are case sensitive. In example above placed real world http
+     * information that output in logs, meanwhile in code headers could be compared to
+     * names with different case. That is why in tests .toUpperCase() are used.
+     * For example, in console you get output as 'x-helpscout-event', in documentation it is mentioned as
+     * 'X-HelpScout-Event' and in Webhook class it is compared to 'X-HELPSCOUT-EVENT'.
+     */
+    @Test
+    public void basicTestForCustomEventHeader() {
+
+        HttpServletRequest httpServletRequest = mock(HttpServletRequest.class);
+
+        when(httpServletRequest.getHeader("x-helpscout-event".toUpperCase())).thenReturn("customer.created");
+
+        Webhook webhook = new Webhook("SecretKey", httpServletRequest);
+
+        assertThat(webhook.getEventType(), equalTo("customer.created"));
+        assertTrue(webhook.isCustomerEvent());
+    }
+
+    /**
+     * Calculating of signature performed based on JSON data and compared to
+     * data in header 'x-helpscout-signature'.
+     * At file 'webhook_customer' located example structure of json send to webhook.
+     */
+    @Test
+    @SneakyThrows
+    public void basicTestForCheckIfRequestIsValid() {
+
+        HttpServletRequest httpServletRequest = mock(HttpServletRequest.class);
+        BufferedReader bufferedReader = new BufferedReader(new StringReader(readJsonDataToString("webhook_customer")));
+
+        when(httpServletRequest.getReader()).thenReturn(bufferedReader);
+        when(httpServletRequest.getHeader("x-helpscout-signature".toUpperCase())).thenReturn("gV91IzHpvzCSLYW+/QGAxfm7KOM=");
+
+        Webhook webhook = new Webhook("SecretKey", httpServletRequest);
+
+        assertTrue(webhook.isValid());
+    }
+
+    /**
+     * Simple test for getting customer object from Webhook class via JSON data
+     */
+    @Test
+    @SneakyThrows
+    public void basicTestForReadingJsonData() {
+
+        HttpServletRequest httpServletRequest = mock(HttpServletRequest.class);
+        BufferedReader bufferedReader = new BufferedReader(new StringReader(readJsonDataToString("webhook_customer")));
+
+        when(httpServletRequest.getReader()).thenReturn(bufferedReader);
+
+        Webhook webhook = new Webhook("SecretKey", httpServletRequest);
+        Customer customer = webhook.getCustomer();
+
+        assertThat(customer.getFirstName(), equalTo("First_Name"));
+        assertThat(customer.getLastName(), equalTo("Last_Name"));
+        assertThat(customer.getEmails().get(0).getValue(), equalTo("some@mail.com"));
+    }
+
+    @SneakyThrows
+    private static String readJsonDataToString(String jsonFileName) {
+        Path pathToJsonData = Paths.get(ClassLoader.getSystemResource("responses/" + jsonFileName + ".json").toURI());
```

Comment:
I get this with `Path` solution from AbstractApiClientTest (line 41), so seem like it was unnoticed moved to 1.7 java, before my commit :) I will search what can be used instead of Path in both places.